spec/rubocop/cop/style/hash_each_methods_spec.rb

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+
+RSpec.describe RuboCop::Cop::Style::HashEachMethods do
+  subject(:cop) { described_class.new }
+
+  context 'when node matches a keys#each or values#each' do
+    context 'when receiver is a send' do
+      it 'registers an offense for foo#keys.each' do
+        expect_offense(<<-RUBY)
+          foo.keys.each { |k| p k }
+              ^^^^^^^^^ Use `each_key` instead of `keys.each`.
+        RUBY
+      end
+
+      it 'registers an offense for foo#values.each' do
+        expect_offense(<<-RUBY)
+          foo.values.each { |v| p v }
+              ^^^^^^^^^^^ Use `each_value` instead of `values.each`.
+        RUBY
+      end
+
+      it 'does not register an offense for foo#each_key' do
+        expect_no_offenses('foo.each_key { |k| p k }')
+      end
+
+      it 'does not register an offense for Hash#each_value' do
+        expect_no_offenses('foo.each_value { |v| p v }')
+      end
+
+      it 'auto-corrects foo#keys.each with foo#each_key' do
+        new_source = autocorrect_source('foo.keys.each { |k| p k }')
+        expect(new_source).to eq('foo.each_key { |k| p k }')
+      end
```

Comment:
Can you use `expect_correction` for test an automatic correction? The following address will be helpful.
https://github.com/rubocop-hq/rubocop/pull/7650